sms_notifservice.py

```diff
@@ -108,27 +105,45 @@ def _prepareNotifQueue(self, notif_queue):
 
             for notif_queue_rec in notif_queue:
-                for user in users:
-                    if notif_queue_rec.userId == user["userId"]:
-                        if self._phone_number_field_name:
-                            tid = user["trackorId"]
-                            notif_queue_rec.trackor_id = tid
-                            try:
-                                notif_queue_rec.phone_number = self._user_trackor.get_phone_number_by_field_name_and_trackor_id(
-                                    self._phone_number_field_name,
-                                    tid)
-                            except Exception as e:
-                                self._integrationLog.add(LogLevel.ERROR, "Can't get Phone Number from User Trackor. Notif "
-                                                                        "Queue ID = [{}]".format(str(notif_queue_rec.notifQueueId)), str(e))
-                        else:
-                            notif_queue_rec.phone_number = user["phoneNumber"]
-
+                if notif_queue_rec.userId is not None:
+                    notif_queue_rec.phone_number = self._get_phone_number(users[notif_queue_rec.userId])
 
         return notif_queue
 
+    def _get_phone_number(self, user):
+        if self._phone_number_field_name:
+            try:
+                phone_number = self._user_trackor.get_phone_number_by_field_name_and_trackor_id(
+                    self._phone_number_field_name,
+                    user["trackorId"])
+            except Exception as e:
+                self._integrationLog.add(LogLevel.ERROR, "Can't get Phone Number from User Trackor. "
+                                                         "Trackor ID = [{}]".format(str(user["trackorId"])),
+                                        str(e))
+        else:
+            phone_number = user["phoneNumber"]
+
+        return phone_number
+    
+    def _format_phone_number(self, phone_number):
+        formatted_phone_number  = phone_number.replace("-", "")
+        formatted_phone_number = formatted_phone_number.replace("(", "").replace(")", "")
+        if formatted_phone_number[0] != "+":
+            formatted_phone_number = "+1" + formatted_phone_number
+            self._integrationLog.add(LogLevel.WARNING,
+                                     "Warning! +1 prefix added to phone number. Final value: [{}]".format(formatted_phone_number))
+        
+        return formatted_phone_number
+    
+    def _check_phone_number(self, phone_number, formatted_phone_number):
+        if re.fullmatch("\+\d{2,15}", formatted_phone_number) is None:
+            self._integrationLog.add(LogLevel.ERROR,
+                                     "Incorrect Phone Number format.",
+                                     "Original Phone Number: [{0}]\nProcessed Phone Number: [{1}]".format(phone_number, formatted_phone_number))
+            raise Exception("Incorrect Phone Number format. Phone Number: [{}]".format(phone_number))
```

Comment:
I think we should create a new exception and then use it here.

Reply:
I created an Issue [IHub-217132](https://trackor.onevizion.com/form/ConfigAppForm.do?id=100106481811&ttid=10009161) to fix this in the whole integration